scripts/updates/api/termux_github_api_get_tag.sh

```diff
@@ -1,95 +1,84 @@
 	fi
 
-	local PKG_SRCURL="$1"
-	local TAG_TYPE="${2:-}"
-	local FILTER_REGEX="${3:-}"
+	local project tag_type filter_regex=""
 
-	local project
-	project="$(echo "${PKG_SRCURL}" | cut -d'/' -f4-5)"
-	project="${project#git+}"
+	project="$(cut -d'/' -f4-5 <<<"${TERMUX_PKG_SRCURL}")"
+	tag_type="$TERMUX_PKG_UPDATE_TAG_TYPE"
 
-	if [[ -z "${TAG_TYPE}" ]]; then # If not set, then decide on the basis of url.
-		if [[ "${PKG_SRCURL:0:4}" == "git+" ]]; then
-			# Get newest tag.
-			TAG_TYPE="newest-tag"
+	if [[ -z "${tag_type}" ]]; then # If not set, then decide on the basis of url.
+		if [[ "${TERMUX_PKG_SRCURL:0:4}" == "git+" ]]; then
+			tag_type="newest-tag" # Get newest tag.
 		else
-			# Get the latest release tag.
-			TAG_TYPE="latest-release-tag"
+			tag_type="latest-release-tag" # Get the latest release tag.
 		fi
 	fi
-	if [[ -n "${FILTER_REGEX}" && "${TAG_TYPE}" != "latest-regex" ]]; then
-		termux_error_exit <<-EndOfError
-		ERROR: You can only specify a regex with TAG_TYPE="latest-regex"
-		EndOfError
+
+	if [[ "${tag_type}" == "latest-regex" ]]; then
+		filter_regex="$TERMUX_PKG_UPDATE_VERSION_REGEXP"
 	fi
 
-	local jq_filter
-	local api_url="https://api.github.com"
+	local jq_filter api_url="https://api.github.com"
 	local -a curl_opts=(
 		--silent
 		--location
 		--retry 10
 		--retry-delay 1
+		-A "Termux update checker 1.1 (github.com/termux/termux-packages)"
 		-H "Authorization: token ${GITHUB_TOKEN}"
 		-H "Accept: application/vnd.github.v3+json"
 		--write-out '|%{http_code}'
 	)
 
-	if [[ "${TAG_TYPE}" == "newest-tag" ]]; then
+	case "${tag_type}" in
+	newest-tag)
 		# We use graphql intensively so we should slowdown our requests to avoid hitting github ratelimits.
 		sleep 1
 
 		api_url="${api_url}/graphql"
 		jq_filter='.data.repository.refs.edges[0].node.name'
 		curl_opts+=(-X POST)
-		curl_opts+=(
-			-d "$(
-				cat <<-EOF | tr '\n' ' '
-					{
-						"query": "query {
-							repository(owner: \"${project%/*}\", name: \"${project##*/}\") {
-								refs(refPrefix: \"refs/tags/\", first: 1, orderBy: {
-									field: TAG_COMMIT_DATE, direction: DESC
-								})
-								{
-									edges {
-										node {
-											name
-										}
+		curl_opts+=(-d "$(
+			cat <<-EOF | tr '\n' ' '
+				{
+					"query": "query {
+						repository(owner: \"${project%/*}\", name: \"${project##*/}\") {
+							refs(refPrefix: \"refs/tags/\", first: 1, orderBy: {
+								field: TAG_COMMIT_DATE, direction: DESC
+							})
+							{
+								edges {
+									node {
+										name
 									}
 								}
 							}
-						}"
-					}
-				EOF
-			)"
-		)
-
-	elif [[ "${TAG_TYPE}" == "latest-release-tag" ]]; then
+						}
+					}"
+				}
+			EOF
+		)")
+	;;
+	latest-release-tag)
 		api_url="${api_url}/repos/${project}/releases/latest"
 		jq_filter=".tag_name"
-	elif [[ "${TAG_TYPE}" == "latest-regex" ]]; then
+	;;
+
+	latest-regex)
 		api_url="${api_url}/repos/${project}/releases"
```

Comment:
I'm not 100% sure `releases` is the right URI namespace to be looking in here.
When I implemented this for the Gitlab method I found that it works more reliably with the list of tags, since some projects do not use releases.